### PR TITLE
feat: add GetFloatingLicenseMode function

### PR DESCRIPTION
**Pull request by muneebkq**
`GetFloatingLicenseMode` to the codebase. This function allows retrieving the mode of the floating license (online or offline). The function takes a pointer to a buffer and the size of the buffer as parameters. The function returns LF_OK if successful, LF_E_PRODUCT_ID if the product ID is invalid, LF_E_NO_LICENSE if no license is found, and LF_E_BUFFER_SIZE if the buffer size is insufficient.

### CODE REVIEW

File: lexfloatclient.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func floatingLicenseCallbackWrapper(status int) {
 
     PARAMETERS:
     * productId - the unique product id of your application as mentioned
-      on the product page in the dashboard.
+    on the product page in the dashboard.
```
Comment (ahmad-kemsan) [inline lexfloatclient.go]: @muneebkq No need for this change.

File: lexfloatclient.go [hunk 2]
```diff
@@ -125,14 +126,14 @@ func SetFloatingClientMetadata(key string, value string) int {
...
 func GetFloatingClientLibraryVersion(libraryVersion *string) int {
-    var cLibraryVersion = getCArray()
-    status := C.GetFloatingClientLibraryVersion(&cLibraryVersion[0], maxCArrayLength)
-    *libraryVersion = ctoGoString(&cLibraryVersion[0])
-    return int(status)
+	var cLibraryVersion = getCArray()
+	status := C.GetFloatingClientLibraryVersion(&cLibraryVersion[0], maxCArrayLength)
+	*libraryVersion = ctoGoString(&cLibraryVersion[0])
+	return int(status)
 }
```
Comment (ahmad-kemsan) [inline lexfloatclient.go:137]: @muneebkq Is this the UI reflecting this or was the indentation by mistake added here?

File: lexfloatclient.go [hunk 3]
```diff
@@ -322,6 +326,25 @@ func HasFloatingLicense() int {
...
+   PURPOSE: Gets the mode of the floating license (online or offline).
+
+   PARAMETERS:
+   * modePtr - pointer to a buffer that receives the value of the string
+   * length - size of the buffer pointed to by the value parameter
+
+   RETURN CODES: LF_OK, LF_E_PRODUCT_ID, LF_E_NO_LICENSE, LF_E_BUFFER_SIZE
+*/
+
+func GetFloatingLicenseMode(modePtr *string) int {
```
Comment (ahmad-kemsan) [inline lexfloatclient.go]: @muneebkq Extra line between the docs and function. Also here we do not use the Ptr in the param across the go wrappers. Please check the other similar functions

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the comments

File: lexfloatclient.go (hunk 2 above)
Comment (muneebkq) [reply lexfloatclient.go:137]: there was no proper indentation earlier

File: lexfloatclient.go (hunk 3 above)
Comment (muneebkq) [reply lexfloatclient.go]: done

**Review by ahmad-kemsan — APPROVED** [review]

**Pull request merged**